runtime/lua/vim/diagnostic.lua

```diff
@@ -1220,6 +1220,18 @@ end
 ---                      string, it is prepended to each diagnostic in the window with no
 ---                      highlight.
 ---                      Overrides the setting from |vim.diagnostic.config()|.
+---            - suffix: (function, string, or table) Suffix each diagnostic in the floating
+---                      window. If a function, it must have the signature (diagnostic, i,
+---                      total) -> (string, string), where {i} is the index of the diagnostic
+---                      being evaluated and {total} is the total number of diagnostics
+---                      displayed in the window. The function should return a string which
+---                      is appended to each diagnostic in the window as well as an
+---                      (optional) highlight group which will be used to highlight the
+---                      suffix. If {suffix} is a table, it is interpreted as a [text,
+---                      hl_group] tuple as in |nvim_echo()|; otherwise, if {suffix} is a
+---                      string, it is appended to each diagnostic in the window with no
+---                      highlight.
+---                      Overrides the setting from |vim.diagnostic.config()|.
```

Comment:
Because this is almost identical to the `prefix` option, let's just point users to that to avoid duplicating documentation here. We can include a point to clarify that `suffix` appends text while `prefix` prepends it (just in case that's not already obvious from the names).